### PR TITLE
Add test suite to upgrade an RBD mirroring cluster from 4x to 5x

### PR DESCRIPTION
**Pull request by manasagowri**
Signed-off-by: mgowri <mgowri@redhat.com>

Execution logs - https://jenkins.ceph.redhat.com/view/RHCS%20QE/job/rhceph-test-executor/1543/console

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: ceph/ceph.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """This module implements the required foundation data structures for testing."""
 import datetime
 import json
+import pdb
```
Comment (VasishtaShastry) [inline ceph/ceph.py:4]: Can we please remove all pdb imports ?

File: run.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env python
 
 # Allow parallelized behavior of gevent. It has to be the first line.
+import pdb
```
Comment (VasishtaShastry) [inline run.py:4]: Can we please remove pdb imports ?

File: suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+#     Node2 - client
+#     Node3 - OSD
+#     Node4 - OSD,
+#     Node5 - OSD
+#     Node6 - RBD Mirror
+
+# The following evaluations are carried out
+#   (1) Configures RBD Mirroring
+#   (2) Creates Pool Image and enables Mirroring
+#   (3) Runs IO using rbd bench
```
Comment (VasishtaShastry) [inline suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml:23]: Can we please update the steps?

File: suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+# Test-Suite: tier-1_rbd_mirror.yaml
+# Test-Case: Configure RBD Mirror setup and run IOs
+# Polarion ID : CEPH-83573329 - RBD HA MirroringDraft
+#
+# Cluster Configuration:
+#    cephci/conf/nautilus/rbd/tier-1_rbd_mirror.yaml
+#    No of Clusters : 2
+#    Each cluster configuration
+#    6-Node cluster(RHEL-8.3 and above)
+#    1 MONS, 1 MGR, 3 OSD and 1 RBD MIRROR service daemon(s)
```
Comment (VasishtaShastry) [inline suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml:12]: Can we please use 3 mon cluster for upgrade scenarios ?

File: tests/ceph_ansible/add_rbd_mirror_daemon.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    return rc
+
+
+def write_configs_to_rbdmirrors_yaml(ceph_installer, config):
+    """Write contents to rbdmirrors.yaml file."""
+
+    rbdmirrors_yaml = yaml.dump(config.get("ansi_config"), default_flow_style=False)
+
+    log.debug(f"global vars {rbdmirrors_yaml}")
```
Comment (VasishtaShastry) [inline tests/ceph_ansible/add_rbd_mirror_daemon.py:60]: `global vars` might confuse users looking at logs

**Review by VasishtaShastry — CHANGES_REQUESTED** [review]
ceph.py needs to be reviewed after changes.

File: ceph/ceph.py (hunk 1 above)
Comment (manasagowri) [reply ceph/ceph.py]: Done.

File: run.py (hunk 2 above)
Comment (manasagowri) [reply run.py]: Done.

File: suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml (hunk 3 above)
Comment (manasagowri) [reply suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml]: Done.

File: suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml (hunk 4 above)
Comment (manasagowri) [reply suites/pacific/rbd/tier-1_rbd_mirroring_upgrade-4-to-latest.yaml]: Done.

**Review by Manohar-Murthy — APPROVED** [review]
Group review is done.

**Review by VasishtaShastry — APPROVED** [review]

**Pull request merged**